src/lib/bindings/bindingDefinitions.ts

```diff
@@ -78,27 +85,64 @@ export function bindMap(
 ): Array<Binding> {
   const instance = getCurrentComponentInstance();
   if (instance) {
+    // If we pass an array of refs instead of a refCollection,
+    // these are only set up once
     if (Array.isArray(target)) {
       return target.map((ref, index) => bind(ref, getProps(ref, index)));
     }
+
     // target.getRefs() is reactive and triggers this watchEffect to update
     // as soon as the underlying ref array updates when the items in the DOM
     // are updated
-    watchEffect(() => {
-      // TODO: should we check if this item already has bindings attached to it?
-      const bindings = ((target as any).getRefs() as Array<
-        ElementRef<RefElementType, BindProps> | ComponentRef<ComponentFactory<any>>
-      >).map((ref, index) => bind(ref, getProps(ref, index)));
-
-      // TODO: should we register this as part of the component instance
-      //  so they get cleaned up on unmount?
-      applyBindings(bindings, instance);
-    });
+    const disposeWatch = watch(
+      () =>
+        (target as any).getRefs() as Array<
+          ElementRef<RefElementType, BindProps> | ComponentRef<ComponentFactory<any>>
+        >,
+      (refs, oldValue, onInvalidate) => {
+        const bindings = refs.map((ref, index) => bind(ref, getProps(ref, index)));
+
+        const removeBindingList = applyBindings(bindings, instance) || [];
+
+        onInvalidate(() => {
+          removeBindingList.forEach((binding) => binding && binding());
```

Comment:
Would this be possible?
```suggestion
          removeBindingList.forEach((binding) => binding?.());
```